modules/ROOT/pages/deployment/storage/s3.adoc

```diff
@@ -0,0 +1,51 @@
+= S3 Simple Storage Service
+:toc: right
+:toclevels: 2
+:description: Infinite Scale can store blobs on S3 storage. This document gives some general information and considerations for the use of S3.
+
+== Introduction
+
+{description} See the xref:deployment/storage/general-considerations.adoc#s3[General Storage Considerations] in the S3 section for which data is stored where.
+
+== S3 Overview
+
+This section gives a brief description about S3 and how it compares to POSIX based filesystems.
+
+S3 based storage is well positioned for customers expecting a very high capacity demand for storing Infinite Scale blobs.
+
+* In S3, objects are stored by name in a collection called a bucket and have no hierarchical order. +
+Objects in a bucket are indexed and can be listed with a prefix based search. This provides a clear scaling advantage. Compared POSIX, the consistency domain is constrained to a single bucket instead of the entire hierarchical tree.
+
+* The bucket is guaranteed eventually consistent. +
+The application must accept that an object that was just updated may or may not appear instantly in the list and may or may not have the latest information, the platform is allowed a best effort at being immediately consistent. Over time, the performance of the system shows that data is in general fully consistent. The architectural freedom to relax the constraint implies real performance and scalability advantages.
+
+* Object storage works best for static storage. +
+This is especially true for unstructured data, where you write data once but may need to read it many times.
+
+* Requests are REST based and self-contained. +
+This method is very well adapted to WAN based interactions.
+
+* Requests are signed with a shared secret, and communicated over an SSL connection. +
+This provides protection against man in the middle attacks and prevents passwords from being communicated in requests.
+
+For more and additional details on S3 read the https://cloud.google.com/learn/what-is-object-storage[What is Object storage?, window=_blank] from Google.
+
+S3 comparison to POSIX::
+* *Performance:* +
+The POSIX filesystem interface is inherently IOPS centric. There is a lot of communication which is expensive and hard to scale. The RESTful S3 API addresses this by transforming IOPS into a throughput problem. Throughput is easier and cheaper to scale. This is the reason why an object storage has high performance at a massive scale.
+
+* *Semantics:* +
+There is no way to guarantee consistency correctness because object operations are atomic and immutable. As a consequence, uncommitted data can be lost in case of a crash or corruption issues can occur in case of shared shared buckets.
```

Comment:
```suggestion
There is no way to guarantee consistency correctness because object operations are atomic and immutable. As a consequence, uncommitted data can be lost in case of a crash or corruption issues can occur in case of shared buckets.
```